src/test/scala/TranspilerSpec.scala

```diff
@@ -649,6 +649,67 @@ class TranspilerSpec extends AnyFlatSpec {
     avroSchema should equal(expectedRecord)
   }
 
+  it should "properly inline defs given a complex graph of self and circular references" in {
+    val extension = Right(JsonSchema.empty.copy(
+      types = Seq(JsonSchemaArray),
+      items = Seq(Right(JsonSchema.empty.copy(ref =  Uri.parseOption("#/definitions/Extension"))))
```

Comment:
Nit-picky with the whitespace:
```suggestion
      items = Seq(Right(JsonSchema.empty.copy(ref = Uri.parseOption("#/definitions/Extension"))))
```